defaults/main.yml

```diff
@@ -1,70 +1,157 @@
-    netmask6: 64                # Optional if IPv6 not indicated.
-    gw6: 200:db8::1             # Optional.
-    bridge: vmbr0
-    firewall: false             # Optional. # Setting netif_firewall in TRUE, enable the use of firewall on the network interface
-    rate_limit: 5               # Optional. (In MB/s)
-    vlan_tag: 200               # Optional.
-  - id: net1
-    name: eth1
-    ip6: 200:db8::10
-    netmask6: 64
-    bridge: vmbr1
-
-mounts: []  # No aditional mount points by default
-#   - id: mp0
-#     storage: local-lvm
-#     size: 16
-#     mount_point: "/mnt/data"
-#     acl: false                     # Optional.
-#     quota: false                   # Optional.
-#     backup: false                  # Optional.
-#     skip_replication: false        # Optional.
-#     read_only: false               # Optional.
-#   - id: mp1
-#     storage: local-lvm
-#     size: 8
-#     mount_point: "/mnt/logs"
-
-# You can change the timeout for the operations of the module according to the performance of your remote host
-# proxmox_create_lxc_timeout: 30 
+# vars needed to define the parameters for the wraped Proxmox API
+
+#########################################################
+### Proxmox API connection and authentication section ###
+#########################################################
+
+# Various module options used to have default values. This cause problems when user expects different behavior from proxmox
+# by default or fill options which cause problems when they have been set. The default value is compatibility, which will
+# ensure that the default values are used when the values are not explicitly specified by the user.From community.general
+# 4.0.0 on, the default value will switch to no_defaults. To avoid deprecation warnings, please set proxmox_default_behavior
+# to an explicit value. This affects the disk, cores, cpus, memory, onboot, swap, cpuunits options.
+# See https://docs.ansible.com/ansible/latest/collections/community/general/proxmox_module.html#parameter-proxmox_default_behavior
+pve_lxc_proxmox_default_behavior: compatibility
+
+# Proxmox node hostname where we create or manage an LXC container
+pve_node: mynode
+
+# FQDN or IP of the Proxmox API endpoint where we manage the cluster or node
+pve_api_host: mynode.mycluster.org
+
+# User to use to connect to the Proxmox cluster API
+pve_api_user: automations@pam    # Optional if token based authentication is used
+# Password for the previous API user (BETTER PUT THIS IN A VAULT, this dummy example can cause security issues)
+pve_api_password: PaSsWd_f0r-AuToMaTi0nS    # Optional if token authentication is used
+
+# pve_api_token_id: automations@pam!ansible                       # Optional if user-password based authentication is used
+# pve_api_token_secret: 0b029a23-1ca3-a93f-8d90-5a4c9d064199      # Optional if user-password based authentication is used
```

Comment:
I tried to use token authentication but got this error:
```
authorization on proxmox cluster failed with exception: __init__() got an unexpected keyword argument 'token_value'
```
I tried to figure it out but failed. Ansible module fails on [this line](https://github.com/ansible-collections/community.general/blob/main/plugins/modules/cloud/misc/proxmox_template.py#L223)

Reply:
This also seems to be purely due to an error in the module's implementation (role simply acts as a gateway from the  variables `pve_api_token_id` and `pve_api_token_secret` to the module's `api_token_id` and `api_token_secret` parameters respectively).

I'll check [the collection repository](https://github.com/ansible-collections/community.general) for any reports on this issue. Otherwise I will report a new one.